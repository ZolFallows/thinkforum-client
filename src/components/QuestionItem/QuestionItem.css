code {
    background-color: #eee;
    border-radius: 3px;
    font-family: courier, monospace;
    padding: 0 3px;
}

.Question_box {
    border: 1px solid #ececec;
    background: white;
    padding: 10px;
}
.Question_info {
    padding: 15px 0 0 0;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
.date_number {
    display: flex;
    flex-direction: row;
}
.Posted_date, .Answer_number {
    display: flex;
    align-self: flex-end;
    font-size: 0.8rem;
    color: grey;
}