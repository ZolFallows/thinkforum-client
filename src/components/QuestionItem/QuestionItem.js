import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import Moment from 'react-moment'
import parse from 'html-react-parser'
import Showdown from "showdown"
import './QuestionItem.css'

export default class QuestionItem extends Component {
    constructor(props) {
        super(props)
        this.converter = new Showdown.Converter({
          tables: true,
          simplifiedAutoLink: true
        })
      }

    renderQuestion(question) {
        return (
            <div className="Question_box">
                <div className="User">
                    <h3>
                        <Link to={`/user/${question.user.id}`}>{question.user.full_name}</Link>
                    </h3>
                </div>
                <div className="Question_item">
                    <h3>{question.title}</h3>
                    <div className="Content">
                        {parse(this.converter.makeHtml(question.content))}
                    </div>
                    <div className="Question_info">
                        <div className="Tags">
                            {question.tags.split(',').map(tag => <span key={tag} className="tag">{tag} </span>)}
                        </div>
                        <div className="date_number">
                            <p className="Posted_date"><Moment fromNow>{question.date_created}</Moment></p>
                            &nbsp;&nbsp;&nbsp;
                            <p className="Answer_number">{question.number_of_answers}&nbsp;respond(s)</p>  
                        </div>
                        
                    </div>
                </div>
            </div>
        )
    }

    render() {
        const { question } = this.props
        return (<>
                    {question ? this.renderQuestion(question) : ''}
                </>
          )
    }
}