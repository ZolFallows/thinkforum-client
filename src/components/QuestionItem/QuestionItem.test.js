import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import QuestionItem from './QuestionItem'

const sectionProp = {
                        id: 7,
                        title: "Another discussion",
                        content: "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident.",
                        tags: "react",
                        date_created: "2019-03-27T20:27:19.481Z",
                        user: {
                            id: 1,
                            user_name: "jones",
                            full_name: "John Jones",
                            date_created: "2019-03-27T20:27:19.481Z"
                        },
                        number_of_answers: 4
                    }


describe(`QuestionItem Component`, () => {

    it('renders witout crashing', () => {
        shallow(<QuestionItem />)  
    })

    it('renders empty given no QuestionItem without errors', () => {
        const wrapper = shallow(<QuestionItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders given QuestionItem without errors', () => {
        const wrapper = shallow(<QuestionItem question={sectionProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})
