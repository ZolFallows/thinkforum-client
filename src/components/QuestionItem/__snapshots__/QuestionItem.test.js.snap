// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionItem Component renders empty given no QuestionItem without errors 1`] = `<Fragment />`;

exports[`QuestionItem Component renders given QuestionItem without errors 1`] = `
<Fragment>
  <div
    className="Question_box"
  >
    <div
      className="User"
    >
      <h3>
        <Link
          replace={false}
          to="/user/1"
        >
          John Jones
        </Link>
      </h3>
    </div>
    <div
      className="Question_item"
    >
      <h3>
        Another discussion
      </h3>
      <div
        className="Content"
      >
        <p>
          At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident.
        </p>
      </div>
      <div
        className="Question_info"
      >
        <div
          className="Tags"
        >
          <span
            className="tag"
            key="react"
          >
            react
             
          </span>
        </div>
        <div
          className="date_number"
        >
          <p
            className="Posted_date"
          >
            <t
              ago={false}
              calendar={false}
              decimal={false}
              element="time"
              filter={[Function]}
              fromNow={true}
              interval={60000}
              onChange={[Function]}
              titleFormat=""
              toNow={false}
              unit={null}
              unix={false}
              utc={false}
              withTitle={false}
            >
              2019-03-27T20:27:19.481Z
            </t>
          </p>
             
          <p
            className="Answer_number"
          >
            4
             respond(s)
          </p>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
